backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java

```diff
@@ -150,16 +157,55 @@ public LicenseInfoParsingResult getLicenseInfoForRelease(Release release) throws
     }
 
     @Override
-    public String getLicenseInfoFileForProject(String projectId, User user) throws TException {
+    public String getLicenseInfoFileForProject(String projectId, User user, String outputGeneratorClassName) throws TException {
         assertId(projectId);
         Project project = projectDatabaseHandler.getProjectById(projectId, user);
         assertNotNull(project);
 
         Collection<LicenseInfoParsingResult> projectLicenseInfoResults = getAllReleaseLicenseInfos(projectId, user);
+        for (OutputGenerator generator : outputGenerators) {
+            if (outputGeneratorClassName.equals(generator.getClass().getName())) {
+                return (String) generator.generateOutputFile(projectLicenseInfoResults, project.getName());
+            }
+        }
+        throw new TException("Unknown output generator for String output: " + outputGeneratorClassName);
+    }
+
+    @Override
+    public ByteBuffer getLicenseInfoFileForProjectAsBinary(String projectId, User user, String outputGeneratorClassName) throws TException {
+        assertId(projectId);
+        Project project = projectDatabaseHandler.getProjectById(projectId, user);
+        assertNotNull(project);
 
-        return generateLicenseInfoFile(projectLicenseInfoResults);
+        Collection<LicenseInfoParsingResult> projectLicenseInfoResults = getAllReleaseLicenseInfos(projectId, user);
+        for (OutputGenerator generator : outputGenerators) {
+            if (outputGeneratorClassName.equals(generator.getClass().getName())) {
+                return ByteBuffer.wrap((byte[]) generator.generateOutputFile(projectLicenseInfoResults, project.getName()));
+            }
+        }
+        throw new TException("Unknown output generator for binary output: " + outputGeneratorClassName);
     }
 
+    @Override
+    public List<OutputFormatInfo> getPossibleOutputFormats() {
+        List<OutputFormatInfo> outputPossibilities = new ArrayList<>();
+        for (OutputGenerator generator : outputGenerators) {
+            outputPossibilities.add(
+                    generator.getOutputFormatInfo()
+            );
+        }
+        return outputPossibilities;
+    }
+
+    @Override
+    public OutputFormatInfo getOutputFormatInfoForGeneratorClass(String generatorClassName) throws TException{
+        for (OutputGenerator generator : outputGenerators) {
+            if (generatorClassName.equals(generator.getClass().getName())) {
```

Comment:
This loop and comparison of generator's class name occurs three times in this file. I think it's time to introduce a method `findGeneratorByClassName` or something like that.